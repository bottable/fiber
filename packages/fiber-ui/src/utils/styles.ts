import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
