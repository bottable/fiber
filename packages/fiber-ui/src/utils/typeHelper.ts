export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
