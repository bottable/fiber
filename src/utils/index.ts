export * from './typeHelper'
export * from './mergeProps'
