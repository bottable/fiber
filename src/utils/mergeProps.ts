import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
