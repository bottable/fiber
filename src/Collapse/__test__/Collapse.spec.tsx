import { Collapse } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Collapse', () => {
  it('renders correctly', () => {
    const collapse = render(<Collapse />)

    expect(collapse).toMatchSnapshot()
  })
})
