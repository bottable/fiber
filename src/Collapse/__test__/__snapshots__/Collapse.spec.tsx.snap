// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse renders correctly 1`] = `
<div
  className="sc-iCoHVE Yqzkr"
>
  <div
    className="sc-fujyUd gnCwwK"
  >
    <div
      className="sc-pNWxx cMOlRD"
      onClick={[Function]}
    >
      <div
        className="sc-jrsJCI jDKIlx"
      >
        This is panel header 1
      </div>
      <span
        className="sc-iqAbSa ggTPDo"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      <span
        className="sc-kEqYlL bTOwMo"
      />
    </div>
    <div
      className="sc-gtssRu jbXVcs"
      onTransitionEnd={[Function]}
    >
      <div
        className="sc-crzoUp kTJOWz"
      >
        <p>
          Hello
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-fujyUd gnCwwK"
  >
    <div
      className="sc-pNWxx cMOlRD"
      onClick={[Function]}
    >
      <div
        className="sc-jrsJCI jDKIlx"
      >
        This is panel header 2
      </div>
      <span
        className="sc-iqAbSa ggTPDo"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      <span
        className="sc-kEqYlL bTOwMo"
      />
    </div>
    <div
      className="sc-gtssRu jbXVcs"
      onTransitionEnd={[Function]}
    >
      <div
        className="sc-crzoUp kTJOWz"
      >
        <p>
          Hello
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-fujyUd gnCwwK"
  >
    <div
      className="sc-pNWxx cMOlRD"
      onClick={[Function]}
    >
      <div
        className="sc-jrsJCI jDKIlx"
      >
        This is panel header 3
      </div>
      <span
        className="sc-iqAbSa ggTPDo"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      <span
        className="sc-kEqYlL bTOwMo"
      />
    </div>
    <div
      className="sc-gtssRu jbXVcs"
      onTransitionEnd={[Function]}
    >
      <div
        className="sc-crzoUp kTJOWz"
      >
        <p>
          Hello
        </p>
      </div>
    </div>
  </div>
</div>
`;
